### DISCUSSION TITLE
Is it possible to depend on either brew's `docker` or Docker for Mac?

### DISCUSSION THREAD

**Discussion by dserodio** [Homebrew/discussions#389] [Tap maintenance and Homebrew development]
We have an internal tool that depends on Docker, and we created a formula for it. Currently, it `depends_on "docker"`. I'd like to improve it to support either brew's `docker` package (/usr/local/opt/docker) or Docker for Mac (/usr/local/bin/docker -> /Applications/Docker.app/Contents/Resources/bin/docker). Is this possible?

Thanks in advance

**Comment by gromgit** [comment]
As far as I know, alternate dependencies are not supported, for the same reason `brew install` options in core formulae were removed: The CI cost to test all possible combinations gets ridiculous really fast.

If this is an internal-only formula, and you want `brew` to manage the dependencies for you, pick the formula or the cask and be done with it. If you absolutely positively must support both, then I suggest having your tool do the dependency management, e.g. check for `/usr/local/bin/docker` and prompt the user to choose which one to `brew install` if it's missing.

And if you plan to submit it to Homebrew core at some point, I'd stick with `depends_on "docker"`.

**Comment by carlocab** [comment]
I think there was some old functionality that's now been removed (for homebrew-core, at least) related to depending on `osxfuse` or something that might help you here. I tried to track down the PRs where this was discussed, but couldn't find them. I distinctly remember it involved a user asking about how to do something similar with their own `borgbackup` formula that they maintain in their own tap.

**Reply by gromgit** [reply]
The end result of that is this: https://github.com/borgbackup/homebrew-tap/blob/master/Formula/borgbackup-fuse.rb

I guess it should be possible to write a custom Requirement that implements "alternate dependencies", though that won't be accepted on Homebrew core.

**Reply by carlocab** [reply]
Yes, that's the one. Thanks!

**Reply by dserodio** [reply]
Awesome, thanks! This internal tool is proprietary at the moment so we won't submit this formula to Homebrew core in the near future

**Comment by dserodio** [comment]
@nidr0x this would be a great improvement for cliditas' Homebrew formula when you have some spare time 😄
